server/src/collection.rs

```diff
@@ -157,6 +157,29 @@ impl Collection {
         self.alias.lock().await.clone()
     }
 
+    pub async fn set_locked(&self, locked: bool) -> Result<(), ServiceError> {
+        let items = self.items.lock().await;
+
+        for item in items.iter() {
+            if !item.is_locked().await {
+                item.set_locked(true).await?;
+            }
+            let manager = self.manager.lock().await;
+            let signal_emitter = manager.signal_emitter(self.path.clone())?;
+            Self::item_changed(&signal_emitter, item.path()).await?;
```

Comment:
Why do you call item_changed if you don't change the item state? that sounds useless.

Reply:
The item_changed should be called form the Item. As I stated multiple times :)

Reply:
My bad :) Since, we define the item_changed and it's part of the Secret Collection interface, doesn't make sense to call it inside Collection? Same for calling collection_changed inside Service?

Reply:
Where it is deifned has nothing to do with where it can/should be called.